photon-ml/src/main/scala/com/linkedin/photon/ml/avro/data/NameAndTerm.scala

```diff
@@ -18,13 +18,21 @@ package com.linkedin.photon.ml.avro.data
  * A compact way to represent the feature key as (name, term) pair.
  * @author xazhang
  */
-//TODO: Change the scope of this class and all functions in the companion object to [[com.linkedin.photon.ml.avro]] after Avro related classes/functons are decoupled from the rest of code
+//TODO: Change the scope of this class and all functions in the companion object to [[com.linkedin.photon.ml.avro]]
+//after Avro related classes/functons are decoupled from the rest of code
 protected[ml] case class NameAndTerm(name: String, term: String) {
 
   override def hashCode: Int = {
     (name + NameAndTerm.DELIMITER + term).hashCode
   }
 
+  override def equals(that: Any): Boolean = {
+    that match {
+      case other: NameAndTerm => hashCode == other.hashCode
```

Comment:
@XianXing is definitely right. This `equals` method has the potential to cause confusion at best, data corruption at worst. (Collisions aren't that uncommon either: `"Aa"` and `"BB"` have the same hash code.)